src/python/grpcio_reflection/grpc/reflection/v1alpha/reflection.py

```diff
@@ -0,0 +1,138 @@
+#     * Redistributions in binary form must reproduce the above
+# copyright notice, this list of conditions and the following disclaimer
+# in the documentation and/or other materials provided with the
+# distribution.
+#     * Neither the name of Google Inc. nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+"""Reference implementation for health checking in gRPC Python."""
+
+import threading
+
+import grpc
+from google.protobuf import descriptor_pb2
+from google.protobuf import descriptor_pool
+
+from grpc.reflection.v1alpha import reflection_pb2
+
+
+class ReflectionServicer(reflection_pb2.ServerReflectionServicer):
+  """Servicer handling RPCs for service statuses."""
+
+  def __init__(self, service_names, pool=descriptor_pool.Default()):
+    """
+    Arguments:
+      service_names: Iterable of fully-qualified service names available.
+    """
+    self._service_names = list(service_names)
+    self._pool = pool
+
+  def ServerReflectionInfo(self, requests, context):
+    for request in requests:
+      if request.HasField('file_by_filename'):
+        yield self._file_by_filename(request.file_by_filename)
+      elif request.HasField('file_containing_symbol'):
+        yield self._file_containing_symbol(request.file_containing_symbol)
+      elif request.HasField('file_containing_extension'):
+        yield self._file_containing_extension(
+            request.file_containing_extension.containing_type,
+            request.file_containing_extension.extension_number)
+      elif request.HasField('all_extension_numbers_of_type'):
+        yield self._all_extension_numbers_of_type(
+            request.all_extension_numbers_of_type)
+      elif request.HasField('list_services'):
+        yield self._list_services()
+      else:
+        yield reflection_pb2.ServerReflectionResponse(
+            error_response=reflection_pb2.ErrorResponse(
+                error_code=grpc.StatusCode.INVALID_ARGUMENT.value[0],
+                error_message=grpc.StatusCode.INVALID_ARGUMENT.value[1].encode(),
+            )
+        )
+
+  def _not_found_error(self):
```

Comment:
Because of a unification of "don't use `@staticmethod`" and "functionality shouldn't be located outside the least-general scope they're meant to be used from". Not that that's the policy we follow, but, y'know, mistakes were made.